modules/endpoint-group/main.tf

```
@@ -1,5 +1,20 @@
 locals {
-  enabled = module.this.enabled
+  enabled                 = module.this.enabled
+  endpoint_configurations = try(var.config.endpoint_configuration, toset([]))
+  lb_names                = { for index, configuration in local.endpoint_configurations : index => try(configuration.endpoint_lb_name, null) }
+  eip_addresses           = { for index, configuration in local.endpoint_configurations : index => try(configuration.endpoint_eip_address, null) }
+}
+
+data "aws_lb" "lb" {
+  for_each = { for index, lb_name in local.lb_names : index => lb_name if lb_name != null }
+
+  name = each.value
+}
+
+data "aws_eip" "eip" {
+  for_each = { for index, lb_name in local.eip_addresses : index => lb_name if lb_name != null }
```

Do not reference `index` in `for_each` as that defeats the purpose of making the entries independent of their place in the list.
```suggestion
  for_each = compact(values(local.eip_addresses))
```

---

Oh that's much cleaner.

---

@Nuru the issue is that the Data resources themselves need an index that corresponds to the index of the endpoint configuration. If we remove some of these values via `compact()`, the indices of the data resources will no be 1-1 with with endpoint configurations.

https://github.com/cloudposse/terraform-aws-global-accelerator/blob/8dc47aeea4d8039c68de4799af27171a5401556c/modules/endpoint-group/main.tf#L37

Maybe there's a cleaner way to do this, but I am not sure yet.

---

Resolved (via other threads)